raiden/blockchain_events_handler.py

```diff
@@ -133,10 +133,14 @@ def handle_channel_new_balance(raiden, event):
         previous_balance = previous_channel_state.our_state.contract_balance
         balance_was_zero = previous_balance == 0
 
-        newbalance_statechange = ContractReceiveChannelNewBalance(
-            channel_identifier,
+        deposit_transaction = TransactionChannelNewBalance(
```

Comment:
by convention *state changes* use the following prefixes:

- Receive
- ContractReceive
- Action

I used the `Transaction` prefix for this *State* following the same rationale, problem is that most state objects use the `State` **postfix**. The question is:

Should we standardize use prefixes for all these values classes, or change the `Transaction` to a postfix?

Reply:
Note 1: We should have a codign document where these conventions would be noted down. 

Question: I don't get why we need two different state changes here. Isn't essentially `ContractReceiveChannelNewBalance` the deposit state change?

Here you make this new `TransactionChannelNewBalance` state change and make it a part of `ContractReceiveChannelNewBalance`. Why? What do we gain from this additional complexity? It's not immediately clear to me from reading the code. Can you maybe explain it a bit?

Reply:
> We should have a codign document where these conventions would be noted down

It's part of the base classes docstring:
https://github.com/raiden-network/raiden/blob/master/raiden/transfer/architecture.py#L47-L80

> Can you maybe explain it a bit?

I'm avoiding mixing state changes and state objects, `TransactionChannelNewBalance` is a `State` not a `StateChange`. My idea is to snapshot only `State` objects, not `StateChange` nor  `Event` objects.

The benefit is not huge, it's just a clear separation among these object types.

Reply:
I am just not sure of the benefit of adding 1 extra class but anyway. I won't interfere with your design here then.